address-calculator.ts

```diff
@@ -530,17 +579,19 @@ export async function calculateDeterministicAddress(
       );
     }
 
-    const data = (await response.json()) as ApiResponse;
+    const data = (await response.json()) as AddressesResponse;
 
     if (!data.addresses || data.addresses.length === 0) {
       throw new BitcoinAddressError("No addresses returned from API");
     }
 
     data.addresses.forEach((address) => {
       if (!address.deprecated || address.deprecated === undefined) {
```

Comment:
this is already sufficient
```suggestion
      if (!address.deprecated) {
```